### PR TITLE
Simple revscan more tests

### PR DESCRIPTION
**Pull request by jnfang**
reverseFetch test was modified into the reverseLongerScan()

Added tests for simple reverse scan for:
- Scan on two tables
- Forward scan and reverse scan comparison

### CODE REVIEW

File: test/TestIntegration.java [hunk 1]
```diff
@@ -1004,6 +965,181 @@ public void prefetchMeta() throws Exception {
...
+
+    ArrayList<KeyValue> kvs = rev_rows.get(0); // KV from 'rl3'
+    assertSizeIs(2, kvs);
+    assertEq("v5", kvs.get(0).value());
+    assertEq("v6", kvs.get(0).value());
+
+    kvs = rev_rows.get(1); // KV from 'rl2', qualifer and value will still be in lexico order
+    assertSizeIs(2, kvs);
+    assertEq("v3", kvs.get(0).value());
+    assertEq("v4", kvs.get(0).value());
```
Comment (yimingluci) [inline test/TestIntegration.java]: Why not verify v1 and v4?

**Comment by yimingluci** [comment]
lgtm except one inline comment.

File: test/TestIntegration.java [hunk 2]
```diff
@@ -1004,6 +965,181 @@ public void prefetchMeta() throws Exception {
...
+  @Test
+  public void basicReverseScan() throws Exception {
+    client.setFlushInterval(FAST_FLUSH);
+    final PutRequest put1 = new PutRequest(table, "brs1", family, "q", "v1");
+    final PutRequest put2 = new PutRequest(table, "brs2", family, "q", "v2");
+    final PutRequest put3 = new PutRequest(table, "brs3", family, "q", "v3");
+    Deferred.group(client.put(put1), client.put(put2),
+            client.put(put3)).join();
+    // Scan the same 3 rows created above twice.
+    for (int i = 0; i < 2; i++) {
```
Comment (caprius) [inline test/TestIntegration.java:979]: This function just runs the exact same scan 3 times, right?  What are we trying to test?

File: test/TestIntegration.java [hunk 3]
```diff
@@ -1004,6 +965,181 @@ public void prefetchMeta() throws Exception {
...
+            scanner.nextRows(1).addCallback(new cb()).join();
+        } finally {
+            scanner.close().join();
+        }
+    }
+  }
+
+  /** Longer reverse scan, checks that qualifier and value are still in lexico order. */
+  @Test
+  public void reverseLongerScan() throws Exception {
```
Comment (caprius) [inline test/TestIntegration.java]: Can we make the function names more descriptive of what we're testing?
In this case, it looks like you're testing that the columns are still in forward order.

File: test/TestIntegration.java [hunk 4]
```diff
@@ -1004,6 +965,181 @@ public void prefetchMeta() throws Exception {
...
+    final PutRequest put4 = new PutRequest(table, "rl2", family, "q4", "v4");
+    final PutRequest put5 = new PutRequest(table, "rl3", family, "q5", "v5");
+    final PutRequest put6 = new PutRequest(table, "rl3", family, "q6", "v6");
+
+    Deferred.group(Deferred.group(client.put(put1), client.put(put2),
+                  client.put(put3)),
+          Deferred.group(client.put(put4), client.put(put5),
+                  client.put(put6))).join();
+    final Scanner rev_scanner = client.newScanner(table);
+    rev_scanner.setFamily(family);
```
Comment (caprius) [inline test/TestIntegration.java]: Is there a reason we need to set the family in this test?  I noticed you didn't do it in the previous test.  All the rows have the same family, so I assume it's not necessary?

File: test/TestIntegration.java (hunk 2 above)
Comment (jnfang) [inline test/TestIntegration.java:979]: Yes this function performs the same scan twice. The forward scan test that they had written does the same thing. I think this is to ensure that scanners can be opened on the same rows without error and that the call to nextRows(1) is working in that only one row gets scanned at a time.

File: test/TestIntegration.java (hunk 3 above)
Comment (jnfang) [inline test/TestIntegration.java]: Yes I will rename it reverseRowsOnlyScan

File: test/TestIntegration.java (hunk 4 above)
Comment (jnfang) [inline test/TestIntegration.java]: Yes not necessary. I will remove it.

File: test/TestIntegration.java (hunk 1 above)
Comment (jnfang) [inline test/TestIntegration.java]: No reason. I will add test for v1 and v2 on as well.

**Pull request merged**